assets/tuned/daemon/tuned/profiles/functions/function_set_pstate.py

```diff
@@ -0,0 +1,39 @@
+import os
+import tuned.logs
+from . import base
+from tuned.utils.commands import commands
+
+log = tuned.logs.get()
+
+processor_name = ("sandybridge",
+        "ivybridge",
+        "haswell",
+        "broadwell",
+        "skylake", 
+        "kabylake",
+        "cometlake",
+        "canonlake")
+		
+pmu_path = "/sys/devices/cpu/caps/pmu_name"
+
+class set_pstate(base.Function):
```

Comment:
I would just use a function instead of a full class.

The name should probably be something  like `intel_pstate_by_family` or maybe `intel_recommended_pstate`.

And the documentation should be much more verbose, explaining the rationale.

Reply:
Thank you for the suggestion, i like both the function name, but `intel_recommended_pstate` probably more suitable here.

for the documentation +1

Reply:
Hi @MarSik, 
if i define a function and not class, is it the recommended place to keep them here: https://github.com/redhat-performance/tuned/blob/master/tuned/utils/commands.py

also, i'm asking a very trivial question here:
- Can we reference the function in the openshift-node-performance profile without encapsulating it in a class? For example, [cpulist_invert](https://github.com/openshift/cluster-node-tuning-operator/blob/master/assets/performanceprofile/tuned/openshift-node-performance#L26C33-L26C47) is called in the profile and i assumed it was resolving through [class and its method](https://github.com/openshift/cluster-node-tuning-operator/blob/master/assets/tuned/daemon/tuned/profiles/functions/function_cpulist_invert.py#L8) and not directly calling [the function under utils directory](https://github.com/openshift/cluster-node-tuning-operator/blob/master/assets/tuned/daemon/tuned/utils/commands.py#L403)